src/environmentd/src/http.rs

```diff
@@ -855,28 +852,25 @@ async fn auth(
         // is the client+secret pair. Bearer auth is an existing JWT that must
         // be validated. In either case, if a username was specified in the
         // client cert, it must match that of the JWT.
-        (Some(frontegg), creds) => {
-            let claims = match creds {
-                Credentials::Password { username, password } => {
-                    let ExchangePasswordForTokenResponse { claims, .. } = frontegg
-                        .exchange_password_for_token(&password, username)
-                        .await?;
-                    claims
-                }
-                Credentials::Token { token } => {
-                    let claims = frontegg.validate_access_token(&token, None)?;
-                    claims
-                }
-                Credentials::DefaultUser | Credentials::User(_) => {
-                    return Err(AuthError::MissingHttpAuthentication)
-                }
-            };
-            let (_, external_metadata_rx) = watch::channel(ExternalUserMetadata {
-                user_id: claims.user_id,
-                admin: claims.is_admin,
-            });
-            (claims.email, Some(external_metadata_rx))
-        }
+        (Some(frontegg), creds) => match creds {
+            Credentials::Password { username, password } => {
+                let auth_session = frontegg.authenticate(&username, &password).await?;
+                let user = auth_session.user().into();
+                let external_metadata_rx = Some(auth_session.external_metadata_rx());
+                (user, external_metadata_rx)
+            }
+            Credentials::Token { token } => {
+                let claims = frontegg.validate_access_token(&token, None)?;
+                let (_, external_metadata_rx) = watch::channel(ExternalUserMetadata {
```

Comment:
Will dropping the sender here have the side effect of closing the receiver, which could cause the session code to bail? Or is it pgwire that does the expiry listening, and so http is exempt and it's ok to immediately close the rx?

Reply:
> Or is it pgwire that does the expiry listening, and so http is exempt and it's ok to immediately close the rx?

Yes, exactly! It's fine that the auth session won't be kept alive by the HTTP request because the HTTP request should be short lived anyway.

Reply:
I split this change out into its own PR, and added a comment to that patch addressing this question: https://github.com/MaterializeInc/materialize/pull/25409/files#r1498425287.